lib/sync-srv.js

```diff
@@ -16,6 +16,31 @@ module.exports = function (cfg) {
 
 var syncInited = false;
 
+/**
+ * This is the main export that exposes the fh.sync API.
+ * An instance of the sync API is created when this function is called.
+ * See api.js for where the instance is wired up and exposed as fh.sync.
+ * 
+ * The sync implementation is made up of the following files:
+ * - sync-srv.js (this file)
+ * -- exposes the server-side sync API (e.g. fh.sync.invoke, fh.sync.handleList, fh.sync.globalHandleCreate)
+ * -- exposes the sync endpoints for fh-mbaas-express to mount at /sync/:datasetId
+ * -- handles requests from clients, accepting and processing sync data
+ * -- calls out to the DataSetModel to actually process creates/reads/updates/deletes/lists
+ * -- keeps a record of all DataSet updates (in fhdb) after processing, marking them as failed, applied or collisions
+ * -- responds to the client with details of all processed updates, and their status
+ * - sync-DataSetModel.js
+ * -- Runs the sync loop (with a 500ms interval between the end of 1 loop and start of next)
+ * -- Calls out to the data handlers (in sync-datahander.js by default) if a particular DataSet is scheduled to be synced
+ * - sync-datahandler.js
+ * -- Default data handler implementation for the DataSet Backend. Uses fh.db to store the DataSets in mongodb.
+ * -- Not used if the developer overwrites all the data handlers
+ * - sync-util.js
+ * -- Has utility functions for generating a hash of an object, sorting a json object by key, logging (and a couple of others)
+ * 
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)